chore(hack): add `-o` flag to deploy published operator versions and snapshots (#10404)
cloudnative-pg/cloudnative-pg: contribute/e2e_testing_environment/README.md

| -e    / --engine <CLUSTER_ENGINE>   | Use the specified Kubernetes engine (e.g., `-e k3d`). (Env: `CLUSTER_ENGINE`)                                   |
| -k    / --k8s-version <K8S_VERSION> | Use the specified Kubernetes full version number (e.g., `-k v1.30.0`). (Env: `K8S_VERSION`)                                   |
| -n    / --nodes \<NODES>            | Create a cluster with the required number of nodes. Used only during "create" command. Default: 3 (Env: `NODES`)              |

> | -o    / --operator-deploy-mode \<MODE> | Controls how the operator is deployed. Use `SOURCE` (default) to build from the local worktree, `main` to deploy the latest published manifest, or an exact version such as `1.28.1` to deploy that release. (Env: `OPERATOR_DEPLOY_MODE`) |

Reviewer: ```suggestion
| -s    / --source \<SOURCE> | Controls which release of the operator is deployed. Use `source` (default) to build from the local worktree or an exact version such as `1.28.1` to deploy that release. (Env: `SOURCE`) |
```

Reviewer: changing it to -s "SOURCE" since it is more explanatory to what we are deploying.
